src/ui/ui_transaction_nbgl.c

```diff
@@ -0,0 +1,230 @@
+/*******************************************************************************
+ *   Ledger App - Decred Wallet
+ *   (c) 2022 Ledger
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+#ifdef HAVE_NBGL
+#include <stdbool.h>  // bool
+#include <string.h>   // memset
+
+#include "os.h"
+#include "os_io_seproxyhal.h"
+#include "ux.h"
+#include "nbgl_use_case.h"
+
+#include "ui_main_menu.h"
+#include "ui_transaction.h"
+#include "ui_shared.h"
+#include "btchip_internal.h"
+#include "glyphs.h"
+
+typedef enum {
+    TX_TYPE_SINGLE_OUTPUT = 0,
+    TX_TYPE_FULL_REVIEW,
+    TX_TYPE_FINALIZE,
+    TX_TYPE_SIGN_MESSAGE,
+} transaction_type_t;
+
+typedef struct {
+    const char* reviewStart;
+    const char* reviewCancel;
+    const char* finishOk;
+    const char* finishCancel;
+    const char* choiceCancel;
+    const char* choiceGoBack;
+} messages_t;
+
+static nbgl_layoutTagValueList_t pairList;
+static nbgl_layoutTagValue_t pairs[4];
+static nbgl_pageInfoLongPress_t infoLongPress;
+static transaction_type_t txType;
+static messages_t msgs;
+static char genericText[70];
+static bool reviewStarted;
+
+static void reviewChoice(bool confirm);
+static void rejectChoice(bool confirm);
+static void rejectUseCaseChoice(void);
+static void displayTransaction(void);
+static void reviewStart(void);
+
+static void rejectChoice(bool confirm) {
+    if (confirm) {
+        nbgl_useCaseStatus(msgs.finishCancel, true, ui_idle);
+        txType == TX_TYPE_SIGN_MESSAGE ? io_seproxyhal_touch_message_signature_verify_cancel(NULL)
+                                       : io_seproxyhal_touch_verify_cancel(NULL);
+    } else {
+        if (reviewStarted) {
+            displayTransaction();
+        } else {
+            reviewStart();
+        }
+    }
+}
+
+static void reviewChoice(bool confirm) {
+    if (confirm) {
+        if (txType != TX_TYPE_SINGLE_OUTPUT) {
+            nbgl_useCaseStatus(msgs.finishOk, true, ui_idle);
+        }
+        txType == TX_TYPE_SIGN_MESSAGE ? io_seproxyhal_touch_message_signature_verify_ok(NULL)
+                                       : io_seproxyhal_touch_verify_ok(NULL);
+    } else {
+        rejectUseCaseChoice();
+    }
+}
+
+static void rejectUseCaseChoice(void) {
+    nbgl_useCaseChoice(NULL,
```

Comment:
I think you can use useCaseConfirm here, so you dont need to use msgs.choiceGoBack

Reply:
Done.